test/e2e/autoscaling/dns_autoscaling.go

```diff
@@ -50,15 +52,18 @@ var _ = SIGDescribe("DNS horizontal autoscaling", func() {
 	f := framework.NewDefaultFramework("dns-autoscaling")
 	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
 	var c clientset.Interface
+	var dc dynamic.Interface
 	var previousParams map[string]string
 	var originDNSReplicasCount int
 	var DNSParams1 DNSParamsLinear
 	var DNSParams2 DNSParamsLinear
 	var DNSParams3 DNSParamsLinear
 
 	ginkgo.BeforeEach(func(ctx context.Context) {
-		e2eskipper.SkipUnlessProviderIs("gce", "gke")
 		c = f.ClientSet
+		dc = f.DynamicClient
+		gvr := schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments/kube-dns"}
+		e2eskipper.SkipIfMissingResource(ctx, dc, gvr, metav1.NamespaceSystem)
```

Comment:
this method is not used anywhere AFAIK
```
grep -r SkipIfMissingResource test/
test/e2e/framework/skipper/skipper.go:// SkipIfMissingResource skips if the gvr resource is missing.
test/e2e/framework/skipper/skipper.go:func SkipIfMissingResource(ctx context.Context, dynamicClient dynamic.Interface, gvr schema.GroupVersionResource, namespace string) {
```
I prefer if you modify or duplicate `getDNSReplicas` to skip if the return value is 0

Reply:
I reworked the testing logic